lib/rubocop/cop/rake/class_definition_in_task.rb

```diff
@@ -33,7 +33,7 @@ class ClassDefinitionInTask < Base
 
         def on_class(node)
           return if Helper::ClassDefinition.in_class_definition?(node)
-          return unless Helper::TaskDefinition.in_task_or_namespace?(node)
+          return unless Helper::TaskDefinition.in_namespace?(node)
```

Comment:
Actually this is the wrong test.  It should be "not in task".  If not in task then it is using the Global Ruby Object namespace.  It makes no difference if it is in a namespace or not.

Reply:
Testing if in a namespace here may have been allowing you to avoid a test for the Global Object Space, but that's really what you want.  According to this Cop's intention:

* Class, Module and Method definitions are OK _iff_ defined at the top-level OR inside a task.
* Class, Module, and Method definitions are not OK if defined inside a namespace.

I don't think it can accurately capture that intend without a test for the definitions at the top level of the file.

Reply:
@pboling Sorry, I dropped this completely. Must have slipped through the cracks

> I don't think it can accurately capture that intent without a test for the definitions at the top level of the file.

There are such tests, which were pre-existing, please refer to:

- `does not register an offense to `def` at the top level`
- `allows class definition outside task`

Given this cop's intention is to only highlight "surprising" behavior, I will leave this logic as is.

Though I also see that the _name_ of this cop is in-correct. Not sure how this is normally handled (is it breaking?) but I will push a rename for now.